grommet/grommet
# SelectMultiple - Change select all and clear all buttons to show when searching

#### What does this PR do?
Adjusts the behavior of SelectMultiple to show 'Select All' and 'Clear All' buttons when searching. 

Note: In the original issue it was mentioned that the buttons should be labeled 'Select All Matches' and 'Clear All Matches' when the options are being filtered by a search term. When I tried this the button labels felt too long and they were taking up too much space within the SelectMultiple drop.

This PR also fixes an issue where selected options were bubbling up when searching. Options should only bubble on after the drop is closed an re-opened.

#### Where should the reviewer start?

#### What testing has been done on this PR?

#### How should this be manually tested?
Test with the existing SelectMultiple stories. Open the drop and type in the search. The 'Select All'/'Clear All' button should still be visible.

#### Do Jest tests follow these best practices?


#### Any background context you want to provide?

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/6444
Closes https://github.com/grommet/grommet/issues/6447

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
Yes
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible
